plugins/UFM_NDT_Plugin/ufm_sim_web_service/topo_diff/ndt_infra.py

```diff
@@ -161,6 +167,51 @@ def run_ibdiagnet_verification_command():
     status, cmd_output = execute_generic_command(IBDIAGNET_PORT_VERIFICATION_COMMAND)
     return status
 
+
+def get_mapping_port_labels2port_numbers():
+    '''
+    Return map with mapping between node_guid and port_lable to port number
+    to be used for topoconfig creation
+    '''
+    port_guid_lable_to_port_num = dict()
+    with open(IBDIAGNET_OUT_DB_CSV_FILE_PATH, 'r', encoding="utf-8") as db_csv_file:
+        # just read file until START_PORTS, then take all the lined until not END_PORTS
+        hierarchy_section_started = False
+        port_line_number = 0
+        while line := db_csv_file.readline():
+            if START_PORT_HIERARCHY_INFO_LABEL in line:
+                hierarchy_section_started = True
+                port_line_number += 1
+            elif hierarchy_section_started:
+                if END_PORT_HIERARCHY_INFO_LABEL in line:
+                    break
+                else:
+                    # somehow the output of ibdiagnet in db_csv will be different (versions)
+                    # need to take fields based on format and not on order. So need to get location
+                    # of the field using header names
+                    port_hierarchy_info = line.strip().split(",")
+                    if port_line_number == 1:
+                        #heade line need to set indexes for ports that we want to use
+                        column_index = 0
+                        for header_column_name in port_hierarchy_info:
+                            if header_column_name == NODE_GUID_INDEX_NAME:
+                                node_guid_index = column_index
+                            elif header_column_name == PORT_NUM_INDEX_NAME:
+                                port_num_index = column_index
+                            elif header_column_name == LABEL_INDEX_NAME:
+                                label_index = column_index
+                            column_index += 1
+                    elif port_line_number >= 2:
+                        try:
+                            port_key = "%s___%s" % (port_hierarchy_info[node_guid_index], port_hierarchy_info[label_index].strip("\""))
+                            port_guid_lable_to_port_num[port_key] = int(port_hierarchy_info[port_num_index])
+                        except Exception as e:
+                            logging.error("Failed to convert port Label to port number:%s." % e)
```

Comment:
will the error message show the full device / port description so we can see in the logs all the info for the problematic ports?